src/scrapers/twitter.iced

```diff
@@ -312,35 +312,78 @@ exports.TwitterScraper = class TwitterScraper extends BaseScraper
       @log "null api_url for #{remote_id}/#{username}"
       return cb err, rc
 
-    u = urlmod.parse api_url
-    u.host = u.hostname = 'mobile.twitter.com'
+    # Do not try to fetch api_url which is of form:
+    # "https://twitter.com/tacovontaco/status/673931888088625152"
+    # because Twitter serves a JavaScript-based page there and the contents of
+    # tweet are not available. Instead, construct a new url to oembed API with
+    # the tweet ID. That API doesn't require authentication and claims not to
+    # be rate limited:
+    # https://developer.twitter.com/en/docs/twitter-api/v1/tweets/post-and-engage/api-reference/get-statuses-oembed
+    u = new urlmod.URL('https://api.twitter.com/1/statuses/oembed.json')
+    u.searchParams.set('id', remote_id.toString())
+    # tell twitter not to include their <script> tag in result.
+    u.searchParams.set('omit_script', '1')
+
     new_api_url = urlmod.format u
-    @log "| rewrite #{api_url} -> #{new_api_url} (since twitter was going to redirect us anyways)"
-    api_url = new_api_url
+    @log "| use oembed API #{new_api_url} for tweet at #{api_url} (remote_id=#{remote_id})"
 
     # calls back with a v_code or null if it was ok
-    await @_get_url_body { url : api_url }, defer err, rc, html
+    await @_get_url_body { url : new_api_url }, defer err, rc, body
 
     if rc is v_codes.OK
```

Comment:
there's a lot of logic here and a lot of early returns. I'm considering putting the whole thing inside of `try` block. Let me know if I should redo it in some other way.